src/AutoRollerAdapterDeployer.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.15;
+import { AutoRoller, PeripheryLike, OwnedAdapterLike } from "./AutoRoller.sol";
+
+interface OwnableFactoryLike {
+    function rlvFactory() external view returns (address);
+    function deployAdapter(address, bytes memory) external returns (address);
+}
+
+interface AutoRollerFactoryLike {
+    function create(
+        OwnedAdapterLike,
+        address rewardRecipient,
+        uint256 targetDuration
+    ) external returns (AutoRoller);
+}
+
+contract AutoRollerAdapterDeployer {
+    PeripheryLike internal immutable periphery;
+
+    constructor(address _periphery) {
+        periphery = PeripheryLike(_periphery);
+    }
+
+    /// @notice Deploys an OwnableERC4626Adapter contract
+    /// @param factory The factory
+    /// @param target The target address
+    /// @param data ABI encoded reward tokens address array
+    /// @param rewardRecipient The address of the reward recipient
+    /// @param targetDuration The duration of each series
+    function deploy(address factory, address target, bytes memory data, address rewardRecipient, uint256 targetDuration) external returns (address adapter, AutoRoller autoRoller) {
+        adapter = periphery.deployAdapter(factory, target, data);
+        autoRoller = AutoRollerFactoryLike(OwnableFactoryLike(factory).rlvFactory()).create(OwnedAdapterLike(adapter), rewardRecipient, targetDuration);
```

Comment:
A lot of the `-like` custom defined interfaces in `AutoRoller.sol` are there because doing it that way saves vv slight amounts of contract size (and... we were.... digging deep for savings. it was a dark time). In a contract like this, i'd personally feel fine about importing the contract directly (e.g. the `AutoRollerFactory`) and using it as the interface

Reply:
Also, if we did decide to keep defining the `AutoRollerFactoryLike` interface ourselves here, we could use `address` as the type for that first arg so we wouldn't have to wrap it with `OwnedAdapterLike` :)

Reply:
> it was a dark time
lol

Reply:
> it was a dark time

🤣